### PR TITLE
[baseline] timestamp-impl + commons-executors

### PR DESCRIPTION
**Pull request by gsheasby**
**Goals (and why)**: Cleaning up baseline issues in a spare moment

**Implementation Description (bullets)**: - Baseline for timestamp-impl
- Baseline for commons-executors

**Concerns (what feedback would you like?)**: nothing spectacular. Keep in mind that we want to avoid breaking internal products with these changes.

**Where should we start reviewing?**: wherever, each change is pretty independent

**Priority (whenever / two weeks / yesterday)**: whenever

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/3341)

### CODE REVIEW

File: commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java [hunk 1]
```diff
@@ -432,10 +435,9 @@ public static ScheduledThreadPoolExecutor newScheduledThreadPoolExecutor(int cor
     @SuppressWarnings("DangerousThreadPoolExecutorUsage")
     public static ScheduledThreadPoolExecutor newScheduledThreadPoolExecutor(int corePoolSize,
             ThreadFactory threadFactory, RejectedExecutionHandler handler) {
-        if (corePoolSize == 0) {
-            corePoolSize = 1;
-        }
-        ScheduledThreadPoolExecutor ret = new ScheduledThreadPoolExecutor(corePoolSize, threadFactory, handler) {
+        int positiveCorePoolSize = corePoolSize > 0 ? corePoolSize : 1;
+        ScheduledThreadPoolExecutor ret = new ScheduledThreadPoolExecutor(positiveCorePoolSize, threadFactory,
+                handler) {
```
Comment (jeremyk-91) [inline commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java]: I think this does actually change behaviour (if someone passed in negative values before, the `ScheduledThreadPoolExecutor` constructor would throw, while now we'll silently set it to 1) - given how widely used this class is I'd prefer for us not to change this too much.

File: commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java [hunk 2]
```diff
@@ -490,41 +492,24 @@ public static Runnable wrap(final Runnable runnable) {
         return new ForwardingRunnableFuture<>(wrappedRunnable, rf);
     }
 
-    private static Runnable wrapRunnable(Runnable runnable) {
+    /**
+     * Wraps the given {@code Callable} so that {@link ExecutorInheritableThreadLocal} variables are
+     * propagated through.
+     */
```
Comment (jeremyk-91) [inline commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java]: 👍

File: timestamp-impl/src/test/java/com/palantir/timestamp/PersistentTimestampServiceMockingTest.java [hunk 3]
```diff
@@ -35,23 +30,22 @@
...
     private static final long INITIAL_TIMESTAMP = 12345L;
     private static final long TIMESTAMP = 100 * 1000;
-    private static final TimestampRange SINGLE_TIMESTAMP_RANGE = TimestampRange.createInclusiveRange(TIMESTAMP, TIMESTAMP);
+    private static final TimestampRange SINGLE_TIMESTAMP_RANGE = TimestampRange.createInclusiveRange(TIMESTAMP,
+            TIMESTAMP);
 
     private static final TimestampRange RANGE = TimestampRange.createInclusiveRange(100, 200);
 
     private PersistentTimestamp timestamp = mock(PersistentTimestamp.class);
-    private ExecutorService executor = Executors.newSingleThreadExecutor();
```
Comment (jeremyk-91) [inline timestamp-impl/src/test/java/com/palantir/timestamp/PersistentTimestampServiceMockingTest.java]: Nice. I'm surprised this executor still existed at all - but I guess this is why we have baseline

Comment (gsheasby) [reply timestamp-impl/src/test/java/com/palantir/timestamp/PersistentTimestampServiceMockingTest.java]: This wasn't baseline - I randomly found some unused code while fixing other baseline issues with that file, and followed the [boy scout rule](https://deviq.com/boy-scout-rule/).

File: commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java (hunk 1 above)
Comment (gsheasby) [reply commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java]: Fair, I'll fix this.

File: commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java (hunk 2 above)
Comment (gsheasby) [reply commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java]: Huh, this was just a method reordering, but it seems the methods are rather similar 😂

File: timestamp-impl/src/test/java/com/palantir/timestamp/PersistentTimestampServiceMockingTest.java (hunk 3 above)
Comment (jeremyk-91) [reply timestamp-impl/src/test/java/com/palantir/timestamp/PersistentTimestampServiceMockingTest.java]: Ah right. I always thought 'variable that was never used' was a baseline rule (it's probably just one of the intellij checks)

File: commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java (hunk 2 above)
Comment (jeremyk-91) [reply commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java]: Yep. Realised that after the fact - but still, good that what this method does is explained

**Pull request merged**